### PR TITLE
Add more tab completion tests

### PR DESCRIPTION
**Pull request by daxian-dbw**
## The second round of adding tab completion tests - issue: #4160 
30 tests are added to improve the test coverage in `CompletionAnalysis.cs`

CompletionAnalysis.cs
- [x] cursor position (you have some of these covered, not sure if all)
- [x] filename after redirection
- [x] line continuation
- [x] DSC resource
- [x] Cursor on special tokens, like comma and minus
- [x] using module <module-name/file-completion>
- [x] Enum property value of DSC Resource

### CODE REVIEW

File: test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1 [hunk 1]
```diff
@@ -589,28 +597,153 @@ Describe "TabCompletion" -Tags CI {
...
-            New-Item -Path "$TestDrive\My [Path]" -ItemType Directory > $null
-            New-Item -Path "$TestDrive\My [Path]\test.ps1" -ItemType File > $null
-
-            Push-Location -Path $TestDrive
+            Push-Location -Path $tempDir
         }
 
         AfterAll {
             Pop-Location
+            Remove-Item -Path $tempDir -Recurse -Force
```
Comment (adityapatwardhan) [inline test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: Please all ```-ErrorAction SilentlyContinue```

**Review by adityapatwardhan — APPROVED** [review]
LGTM with minor comment.

File: test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1 [hunk 2]
```diff
@@ -589,28 +597,153 @@ Describe "TabCompletion" -Tags CI {
             $res.GetNextResult($true).CompletionText | Should Be 'Cmdlet'
             $res.GetNextResult($true).CompletionText | Should Be 'Configuration'
         }
+
+        It "Test history completion" {
+            $history = [pscustomobject]@{
+                CommandLine = "Test history completion"
+                ExecutionStatus = "Stopped"
+                StartExecutionTime = "8/11/2017 11:04:10 AM"
+                EndExecutionTime = "8/11/2017 11:05:49 AM"
```
Comment (iSazonov) [inline test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: The dates is not explicitly used but it is culture dependent. Should our tests work under different cultures?

Comment (adityapatwardhan) [inline test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: @daxian-dbw Can you address feedback from @iSazonov

Comment (daxian-dbw) [inline test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: Good catch. Will change to use a `DataTime` object instead.

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
LGTM.

**Pull request merged**